src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -266,7 +325,10 @@ mongoc_cluster_run_command_opquery (mongoc_cluster_t *cluster,
                                     cluster->iov.len,
                                     cluster->sockettimeoutms,
                                     error)) {
-      mongoc_cluster_disconnect_node (cluster, server_id, true, error);
```

Comment:
libmongoc previously distinguished read and write timeouts (and would not invalidate a server for a read timeout). SDAM currently says a post-handshake timeout should not invalidate (and does not distinguish). I've been mostly convinced this is the wrong behavior and changed this to conform to the spec. Though I created https://jira.mongodb.org/browse/DRIVERS-1246 for further discussion.